packages/dds/tree/src/simple-tree/core/unhydratedFlexTree.ts

```diff
@@ -150,11 +150,26 @@ export class UnhydratedFlexTreeNode implements UnhydratedFlexTreeNode {
 	public adoptBy(parent: UnhydratedFlexTreeField, index: number): void;
 	public adoptBy(parent: UnhydratedFlexTreeField | undefined, index?: number): void {
 		if (parent !== undefined) {
-			assert(
-				this.location === unparentedLocation,
-				0x98c /* Node may not be adopted if it already has a parent */,
-			);
 			assert(index !== undefined, 0xa08 /* Expected index */);
+			if (this.location !== unparentedLocation) {
+				throw new UsageError(
+					"A node may not be in more than one place in a tree or in more than one tree",
```

Comment:
```suggestion
					"A node may not be inserted if it's already in a tree",
```
nit: I like the error being in terms of the action that the user did wrong (a bad insertion) rather than the more abstract invariant that was broken. You phrased the other error below in that way and I like it - it sounds more prescriptive and is easier to read, I think.

Reply:
@noencke I think that phrasing doesn't really cover the case like making an array that contains the same node twice, since putting into two places happens at the same time (inside the same api call with no clear ordering), and it wasn't in a tree before.

Reply:
Actually, the case where an array is constructed with duplicate nodes goes through a different flow and has a different error message. Take a look at the updated test.

Reply:
> @noencke I think that phrasing doesn't really cover the case like making an array is constructed that contains the same node twice, since putting into two places happens at the same time (inside the same api call with no clear ordering), and it wasn't in a tree before.

That's a good point, but I don't think users will be confused by that.

Reply:
@CraigMacomber @noencke I went with Noah's suggestion - see my comment above that says there is a different flow when an arrary is constructed with duplicate nodes. Please let me know if this sounds good.